lively.morphic/rendering/renderer.js

```diff
@@ -161,13 +161,17 @@ export default class Renderer {
     }
 
     for (let morph of this.morphsWithStructuralChanges) {
-      this.renderStructuralChanges(morph);
+      morph.withAllSubmorphsDo(m => !this.renderMap.has(m) && this.renderMorph(m));
```

Comment:
This is probably something better suited for a call, but I find this to be dubious...

I assume this is to "make the nodes ready" for any added morphs? In which scenarios was the previous solution problematic? In `renderStructuralChanges`, those are the first two lines:

```
let node = this.getNodeForMorph(morph);
if (!node) node = this.renderMorph(morph);
```
I do not understand what this covers that is not already covered.

Reply:
We can certainly replace
```
morph.withAllSubmorphsDo(m => !this.renderMap.has(m) && this.renderMorph(m))
```
with
```
morph.withAllSubmorphsDo(m => this.renderStructuralChanges(m));
```

However that would be overkill for what we want to achieve here, which is just the "preparation" of nodes. `renderStructuralChanges` does way more than that.

Long story short: Not having all the nodes already initialized before we apply styling changes caused all kinds of issues. One of them was that scrolls did not get set correctly. I will try to review what issues those were in particular later on.

Reply:
> However that would be overkill for what we want to achieve here, which is just the "preparation" of nodes. `renderStructuralChanges` does way more than that.

Okay, that makes sense. From this follows that we can remove the node initialization from `renderStructuralChanges()`, doesn't it? Since we made sure to initialize all nodes beforehand.

Nothing that we need to act on super urgently, but my feeling when working on this was that the node-patching etc. is complex enough as it is, and having the general loop quite tidy made developing and especially debugging quite a bit more enjoyable.

Reply:
Yes we can certainly refactor to separate the node initialization from the structural change application!